### PR TITLE
build: Really make the acpi feature disableable

### PR DESCRIPTION
**Pull request by rbradford**
The command "cargo build --no-default-features" does not recursively
disable the default features across the workspace. Instead add an acpi
feature at the top-level, making it default, and then make that feature
conditional on all the crate acpi features.

Signed-off-by: Rob Bradford <robert.bradford@intel.com>

### CODE REVIEW

**Comment by rbradford** [comment]
@sboeuf please see if this fixes the issue you observed

File: arch/src/x86_64/mod.rs [hunk 1]
```diff
@@ -127,7 +129,7 @@ pub fn configure_system(
     cmdline_size: usize,
     num_cpus: u8,
     setup_hdr: Option<setup_header>,
-    serial_enabled: bool,
+    _serial_enabled: bool,
```
Comment (sboeuf) [inline arch/src/x86_64/mod.rs]: Why did you add an underscore here?

Comment (rbradford) [reply arch/src/x86_64/mod.rs]: Otherwise the "no default" build on travis will fail because that parameter is only used in the acpi feature.
